### DISCUSSION TITLE
Inner join not working, error says to try reloading schema cache

### DISCUSSION THREAD

**Discussion by romlytech** [supabase/supabase#2579] [Questions] [answered]
I am querying my public.profiles table and would like to get the user's email address:

``` js
const { data, error } = await supabase.from("profiles").select("*, users(*)");
```

I keep getting this error:

``` js
{
  hint: 'If a new foreign key between these entities was created in the database, try reloading the schema cache.',
  message: 'Could not find a relationship between profiles and users in the schema cache'
}
```

I restarted my project server. Not sure why this is happening? The foreign key is set up correctly (as far as I can tell).

**Comment by steve-chavez** [comment]
> I restarted my project server. Not sure why this is happening? The foreign key is set up correctly (as far as I can tell).

Can you try doing `NOTIFY ddl_command_end` on the SQL editor? Querying the foreign table should work If there's a foreign key between users and profiles.

**Reply by romlytech** [reply]
Thanks, however, I tried `NOTIFY ddl_command_end` and am still seeing the same error. :(

**Reply by steve-chavez** [reply]
Hm, perhaps you can double check if there's a foreign key between the two tables? Try running this query:

```sql
SELECT
    tc.table_schema, 
    tc.constraint_name, 
    tc.table_name, 
    kcu.column_name, 
    ccu.table_schema AS foreign_table_schema,
    ccu.table_name AS foreign_table_name,
    ccu.column_name AS foreign_column_name 
FROM 
    information_schema.table_constraints AS tc 
    JOIN information_schema.key_column_usage AS kcu
      ON tc.constraint_name = kcu.constraint_name
      AND tc.table_schema = kcu.table_schema
    JOIN information_schema.constraint_column_usage AS ccu
      ON ccu.constraint_name = tc.constraint_name
      AND ccu.table_schema = tc.table_schema
WHERE tc.constraint_type = 'FOREIGN KEY' AND tc.table_name IN ('users', 'profiles');
```

It should show a row(s) if there's indeed a foreign key between those tables. Please share the result here to confirm.

**Reply by romlytech** [reply]
I only get one result, which is another column that has a f_key. Working good.

In table editor, the key shows:
<img width="426" alt="Screen Shot 2021-07-27 at 10 20 02 AM" src="https://user-images.githubusercontent.com/65119459/127170695-ef771cc6-1c7b-4524-8eee-db08ff296f07.png">

On the database tables page, it does not:
<img width="513" alt="Screen Shot 2021-07-27 at 10 20 37 AM" src="https://user-images.githubusercontent.com/65119459/127170733-fa0b18dc-b559-43cf-938d-da3dd8e34f03.png">

Again, it does not show up when I do the SQL query.

**Comment by romlytech** [comment, marked as answer]
For anyone reading... the auth schema is not available via the js client.

**Reply by n1ghtmare** [reply]
Thanks for the update. Do you have a workaround?

**Reply by victor-cano** [reply]
Maybe you can use rpc, define a function on postresql and the call it with the js client.

[Postgres functions: rpc()](https://supabase.com/docs/reference/javascript/rpc)
